code/modules/clothing/under/accessories/medals.dm

```diff
@@ -7,48 +7,88 @@
+	. = ..()
+	if(.)
+		return
+	if(!ishuman(target) || target == user)
+		return
+
+	. = TRUE // no attack chain please
+
+	var/mob/living/carbon/human/distinguished = target
+	var/obj/item/clothing/under/distinguished_uniform = distinguished.w_uniform
+	if(!istype(distinguished_uniform))
+		distinguished.balloon_alert(user, "no uniform to pin on!")
+		return .
+	if(!can_attach_accessory(distinguished_uniform, user))
+		// Check handles feedback messages and etc
+		return .
+
+	user.visible_message(
+		span_notice("[user] tries pin [src] on [distinguished]'s chest."),
 		span_notice("You try to pin [src] on [distinguished]'s chest."),
 	)
 
-	var/input
-	if(!commended)
-		input = tgui_input_text(attacher, "Reason for this commendation? It will be recorded by Nanotrasen.", "Commendation", max_length = 140)
+	commendation_message = tgui_input_text(user, "Reason for this commendation? It will be recorded by Nanotrasen.", "Commendation", max_length = 140)
+	if(!commendation_message || !pin_checks(user, distinguished))
+		return .
+	if(!do_after(user, 2 SECONDS, distinguished, extra_checks = CALLBACK(src, PROC_REF(pin_checks), user, distinguished)))
+		return .
+
+	if(distinguished_uniform.attach_accessory(src, user))
+		user.visible_message(
+			span_notice("[user] pins [src] on [distinguished]'s chest."),
+			span_notice("You pin [src] on [distinguished]'s chest."),
+		)
+	else
+		user.visible_message(
+			span_warning("[user] fails to pin [src] on [distinguished]'s chest, seemingly unable to part with it."),
+			span_warning("You fail to pin [src] on [distinguished]'s chest."),
+		)
+
+	return .
 
-	if(!do_after(attacher, 2 SECONDS, distinguished))
-		return FALSE
+/obj/item/clothing/accessory/medal/attach(obj/item/clothing/under/attach_to, mob/living/attacher)
+	var/mob/living/distinguished = attach_to.loc
+	if(isnull(attacher) || !istype(distinguished) || distinguished == attacher || awarded_to)
+		// You can't be awarded by nothing, you can't award yourseld, and you can't be awarded someone else's medal
+		return ..()
 
-	attacher.visible_message(
-		span_notice("[attacher] pins [src] on [distinguished]'s chest."),
-		span_notice("You pin [src] on [distinguished]'s chest."),
-	)
-	if(!input)
-		return FALSE
+	awarder = attacher.real_name
+	awarded_to = distinguished.real_name
+
+	update_appearance(UPDATE_DESC)
+	SSblackbox.record_feedback("associative", "commendation", 1, list("commender" = "[awarder]", "commendee" = "[awarded_to]", "medal" = "[src]", "reason" = commendation_message))
+	GLOB.commendations += "[awarder] awarded <b>[awarded_to]</b> the <span class='medaltext'>[name]</span>! \n- [commendation_message]"
+	distinguished.log_message("was given the following commendation by <b>[key_name(attacher)]</b>: [commendation_message]", LOG_GAME, color = "green")
+	message_admins("<b>[key_name_admin(distinguished)]</b> was given the following commendation by <b>[key_name_admin(attacher)]</b>: [commendation_message]")
```

Comment:
lol i don't think i've ever seen this message_admins in the old code despite having commendations be awarded, was this broken for the longest time? not really a suggestion/nit to change anything, just observing that something may have been fucked up here previously

Reply:
I simultaneously think you're right but also think there's no way it was broken for all the years